File: Q3/cpp/Cryptography.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;
class Crytography
{
public:
    long long encrypt(vector<int> numbers)
    {
        //long long 값으로 들어오는데 int로 처리하고있으니 정상적인값이 안들어옴.
        //int maximum = 0;
        long long maximum = 0;
        vector<int> temp(numbers);
        for(int i = 0; i<temp.size();i++)
        {
            temp[i] +=1;
            //long long 값으로 들어오는데 int로 처리하고있으니 정상적인값이 안들어옴.
            //int value = 0;
            long long value = 1;
            for(int l = 0; l<temp.size();l++)
            {
                value *= temp[l];
            }
            maximum = max(maximum,value);
            temp[i] -=1;
        }
        return maximum;
    }
};

int main()
{
    vector<int> numbers = {1000,999,998,997,996,995};
    cout<<Crytography().encrypt(numbers)<<endl;
    return 0;
}

File: Q3/python/Cryptography.py
#import sys

class Crytography:
    def __init__(self):
        pass

    def encrypt(self, numbers : list):
        maximum = 0
        for i in range(0, numbers.__len__()):
            numbers[i] = numbers[i] + 1
            value = 1
            for num in numbers:
                value *= num
                #print(sys.getsizeof(value))
                pass
            maximum = max(maximum, value)
            numbers[i] = numbers[i] - 1
        #파이썬에서의 변수의 크기는 미리 크게잡아두고 필요에의해 크거나 작아지거나 할 수 있음.?
        return maximum

numbers = [1000,999,998,997,996,995]
print(Crytography().encrypt(numbers))